src/test/java/com/dev9/presentation/service/ParallelStreamTest.java

```diff
@@ -0,0 +1,56 @@
+package com.dev9.presentation.service;
+
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+
+public class ParallelStreamTest {
+
+    @Test
+    public void test_in_line() {
+
+        final List<Integer> primes = IntStream.range(2, 1_000_000)
+                .boxed()
+                .parallel()
+                .filter(Prime::isPrime)
+                .collect(Collectors.toList());
+
+        assertThat(primes.size()).isEqualTo(78498);
+    }
+
+    @Test
+    public void test_parallel_supplied_forkJoinPool() throws ExecutionException, InterruptedException {
+
+        ForkJoinPool forkJoinPool = new ForkJoinPool(2);
+        List<Integer> primes = forkJoinPool.submit(
+                () ->
+                        IntStream.range(2, 1_000_000)
+                                .parallel()
+                                .boxed()
+                                .filter(Prime::isPrime)
+                                .collect(Collectors.toList())
+                ).get();
+
+        assertThat(primes.size()).isEqualTo(78498);
+    }
+
+
+    public static class Prime {
```

Comment:
Thanks, 
update and push if you want. To be fair I didn't even look at prime implementation. This is really to demo issues with parallel streams